string requestId
string[] dependencies
string servingRr
string originRr
int8 actionCode  # used to send delete or other special status messages without the need to use separate clients for communication. Will need some testing